src/fable-library/Array.fs

```diff
@@ -539,7 +431,7 @@ let choose (chooser: 'T->'U option) (array: 'T[]) ([<Inject>] cons: Cons<'U>) =
         match chooser array.[i] with
         | None -> ()
         | Some y -> pushImpl res y |> ignore
-    if jsTypeof cons = "function"
```

Comment:
@alfonsogarciacaro How can we write test this using non-js code? This is why I added that `callable` to Fable.Core.

Reply:
I used `jsTypeof cons = "function"` because it's what I usually do in JS, but I guess we can add [<AllowNullLiteral>}` to `type Cons` and then just make a null check because we're not going to pass anything through that argument that's not a constructor.

Reply:
Sorry,  I don't understand  add [} to type Cons`. Please explain

Reply:
Will boxing work with a null check?

Reply:
Sorry, I missed a quote and the text was not well formatted 😅 If we add `AllowNullLiteral` attribute we shouldn't need the boxing. I can do that 👍